### DISCUSSION TITLE
Remove useless empty import

### DISCUSSION THREAD

**Discussion by depsimon** [biomejs/biome#2646] [Q&A]
Is there a way to automatically remove useless empty imports?

Here's a [playground](https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB7AH0AIABmAHIAbwBtACAAJwBkACcACgAKAAoAIAAgAA%3D%3D).

Input:

```ts
import {} from 'useless-import'
import { util } from 'my-util'


export function myUtil() {
  return util()
}
```

Expected output:

```ts
import { util } from 'my-util'


export function myUtil() {
  return util()
}
```

**Comment by Conaclos** [comment]
You can use the linter rule [noUnusedImports](https://biomejs.dev/linter/rules/no-unused-imports/). The rule has a safe fix that remove unused imports.

**Reply by depsimon** [reply]
This rule is enabled and works for unused imports, but doesn't delete the empty imports `{}`.

**Reply by Conaclos** [reply]
How did you end with empty named imports?

We could extend [noUnusedImports](https://biomejs.dev/linter/rules/no-unused-imports/) to support this case. Or this could be part of the import organizer. Not sure.

**Reply by depsimon** [reply]
I think it comes from the VS Code extension that breaks a lot of refactorings. To fix most of them we mainly Undo (Cmd+Z) after fixing with Biome until the result is valid.

**Reply by MYKEU** [reply]
For me I appear to only get an empty import if there are more than one unused named imports; ie:

```tsx
import { Day, format } from 'date-fns'

// becomes

import {} from 'date-fns'
```

Whereas a single unused import correctly removes it entirely

**Reply by wottpal** [reply]
Did someone manage to remove `{}` imports?

**Reply by AlfianChabib** [reply]
> You can use the linter rule [noUnusedImports](https://biomejs.dev/linter/rules/no-unused-imports/). The rule has a safe fix that remove unused imports.

<img width="344" height="119" alt="image" src="https://github.com/user-attachments/assets/9a9bfc37-7af5-4dec-bd6a-8aa2cf8152ab" />
like this
